pnc_cli/user_config.py

```diff
@@ -70,25 +71,34 @@ def __setstate__(self, state):
         else:
             self.keycloak_config = saved_kc_config
 
+
         # if more than a day has passed since we saved the token, and keycloak server configuration is not modified,
         # get a new one
-        if not newtoken and utils.current_time_millis() - self.token_time > 86400000:
+        if not newtoken and utils.current_time_millis() - self.refresh_token_time > 8640000:
             # input the password again since we no longer cache it
             logging.info("Keycloak token has expired for user {}. Retrieving new token...\n".format(self.username))
             newtoken = True
 
+        # if access_token is more than hour old, refresh it
+        if not newtoken and 8640000 >= utils.current_time_millis() - self.access_token_time > 360000:
+            logging.info("Refreshing access token for user {}. \n".format(self.username))
+            refreshtoken = True
+
+        if refreshtoken:
+            self.access_token = self.refresh_access_token()
+
         if newtoken:
             # if using client auth, we simply get a new token.
-            if (self.keycloak_config.client_mode in ['True', 'true', '1']):
-                self.token = self.retrieve_keycloak_token()
+            if self.keycloak_config.client_mode in trueValues:
+                self.access_token = self.retrieve_keycloak_token()
             else:
                 # enter password to get new token, but only if the user has not entered a password in pnc-cli.conf
                 password = self.load_password_from_config(config)
                 if password:
                     self.password = password
                 else:
                     self.password = self.input_password()
-                self.token = self.retrieve_keycloak_token()
+                self.access_token = self.retrieve_keycloak_token()
```

Comment:
When you are setting the refresh token inside the method, you can set there the acess_token as well, no?

Reply:
That was the method's original functionality (that it returns access_token). Same thing is happening in refresh_access_token(self) method. So, this change should be made, I'd rather do it in both retrieve and refresh methods for the sake of consistency.